dali/python/backend_impl.cc

```diff
@@ -208,6 +223,46 @@ void ExposeTensor(py::module &m) {
       )code");
 
   py::class_<Tensor<GPUBackend>>(m, "TensorGPU")
+    .def(py::init([](const py::object object, string layout = "") {
+          PyObject *p_ptr = object.ptr();
+          if (!PyObject_HasAttr(p_ptr, PyUnicode_FromString("__cuda_array_interface__"))) {
+            DALI_FAIL("Provided object doesn't support cuda array interface protocol")
+          }
+          PyObject *ptr_intf = PyObject_GetAttr(p_ptr,
+                                                PyUnicode_FromString("__cuda_array_interface__"));
+          if (ptr_intf == Py_None) {
+            DALI_FAIL("Provided object doesn't support cuda array interface protocol.")
+          }
+          py::dict cu_a_interface = py::reinterpret_borrow<py::dict>(ptr_intf);
+
+          DALI_ENFORCE(cu_a_interface.contains("typestr") && cu_a_interface.contains("shape") &&
+                       cu_a_interface.contains("data"), "Provided object doesn't have required "
+                       "cuda array interface protocol fields.");
+          if (cu_a_interface.contains("strides")) {
+            try {
+              cu_a_interface["strides"].cast<py::none>();
+            } catch (...) {
+               DALI_FAIL("Strided data not supported");
+            }
```

Comment:
Please change it unless there's absolutely no other way to check for `none` than using exceptions...

Reply:
I refered to https://github.com/pybind/pybind11/issues/649, but now I see that I can just call `is_none`.